BetaCameras/BaslerACA/BaslerACA.cs

```diff
@@ -0,0 +1,147 @@
+        }
+
+        ~BaslerACA()
+        {
+            Terminate();
+        }
+
+        protected override void LoadAllAvailableChannels()
+        {
+            ChannelRegistry cr = ChannelRegistry.Instance;
+            Channels.Clear();
+
+            Channels.Add(cr.RegisterChannel(ChannelNames.Color));
+        }
+
+        protected override void ConnectImpl()
+        {
+            List<DeviceEnumerator.Device> list = DeviceEnumerator.EnumerateDevices();
+            DeviceEnumerator.Device dev = null;
+
+            if (!String.IsNullOrEmpty(SerialNumber))
+            {
+                dev = list.Where(d => d.SerialNumber == SerialNumber).First();
+                if(null == dev)
+                {
+                    string msg = string.Format("BaslerACA: no device with s/n {0} found.", SerialNumber);
+                    log.Error(msg);
+                    throw new InvalidOperationException(msg);
+                }
+            }
+            else
+            {
+                dev = list.Where(d => d.Model.StartsWith(DeviceClass)).First();
+                if (null == dev)
+                {
+                    string msg = string.Format("BaslerACA: no device of type {0} found.", DeviceClass);
+                    log.Error(msg);
+                    throw new InvalidOperationException(msg);
+                }
+            }
+
+            _imageProvider.Open(dev.Index);
+
+            if (ActiveChannels.Count == 0)
+            {
+                ActivateChannel(ChannelNames.Color);
+            }
+        }
+
+        protected override void DisconnectImpl()
+        {
+            _imageProvider.Close();
+        }
+
+        protected override void UpdateImpl()
+        {
+            _tsc = new TaskCompletionSource<bool>();
+            _imageProvider.OneShot();
+            _tsc.Task.Wait();
+        }
+
+        protected override CameraImage CalcChannelImpl(string channelName)
+        {
+            switch (channelName)
+            {
+                case ChannelNames.Color:
+                    return new ColorCameraImage(_bitmap);
+            }
+
+            log.Error("Unexpected ChannelName in CalcChannel().");
+            return null;
+        }
+
+        private void OnImageReadyEventCallback()
+        {
+            ImageProvider.Image image = _imageProvider.GetLatestImage();
+
+            if (BitmapFactory.IsCompatible(_bitmap, image.Width, image.Height, image.Color))
+            {
+                BitmapFactory.UpdateBitmap(_bitmap, image.Buffer, image.Width, image.Height, image.Color);
+            }
+            else
+            {
+                if (_bitmap != null)
+                {
+                    _bitmap.Dispose();
+                }
+
+                BitmapFactory.CreateBitmap(out _bitmap, image.Width, image.Height, image.Color);
+                BitmapFactory.UpdateBitmap(_bitmap, image.Buffer, image.Width, image.Height, image.Color);
```

Comment:
This line is in both branches -> move it outside the conditional (then remove the if-branch)